File: src/09.tree/35.606.go
package tree

import (
	"strconv"
)

func Tree2str(root *TreeNode) string {
	var dfs func(node *TreeNode)
	ans := ""

	dfs = func(node *TreeNode) {
		ans += "(" + strconv.Itoa(node.Val)

		if node.Left != nil {
			dfs(node.Left)
		} else if node.Right != nil {
			ans += "()"
		}

		if node.Right != nil {
			dfs(node.Right)
		}

		ans += ")"
	}

	dfs(root)

	return ans[1 : len(ans)-1]
}

File: src/09.tree/35.606.ts
import { TreeNode } from "./01.100";

export function tree2str(root: TreeNode | null): string {
    let ans = ""

    const dfs = (node: TreeNode | null) => {
        ans += "(" + node.val
        
        if (node.left) {
            dfs(node.left)
        } else if (node.right) {
            ans += "()"
        }

        if (node.right) {
            dfs(node.right)
        }

        ans += ")"
    }

    dfs(root)

    return ans.slice(1, -1)
}
